### PR TITLE
Feature/refctoring

### PR DESCRIPTION
**Pull request by teone**
@SeanCondon I am starting a bit of refactoring on the GUI, mainly removing the duplicate attributes that we now have in the base-class.

Not urgent but if you have time let me know what you think (more to come in the next few days)

### CODE REVIEW

File: src/app/roc-edit-base.ts [hunk 1]
```diff
@@ -44,24 +44,26 @@ export abstract class RocEditBase<
...
         protected snackBar: MatSnackBar,
         protected bs: BasketService,
         protected route: ActivatedRoute,
         protected router: Router,
-        protected pathRoot: RocElement,
-        protected pathListAttr: string,
-        protected idAttr: string = 'id',
         public ds: T,
-        protected modelPath: string[],
+        public modelPath: string[],
```
Comment (SeanCondon) [inline src/app/roc-edit-base.ts:59]: looks good

File: src/app/roc-edit-base.ts [hunk 2]
```diff
@@ -44,24 +44,26 @@ export abstract class RocEditBase<
     public unknownSite = 'unknownsite';
     public datasource: T;
 
+    /**
+     * @param snackBar
+     * @param bs
+     * @param route
+     * @param router
```
Comment (SeanCondon) [inline src/app/roc-edit-base.ts:51]: should `modelPath` be in this list?

File: src/app/roc-edit-base.ts [hunk 3]
```diff
@@ -149,14 +155,12 @@ export abstract class RocEditBase<
...
+
         fullPath +=
-            '/' +
-            this.pathListAttr +
-            '[' +
-            this.idAttr +
-            '=' +
-            paramMap.get('id') +
-            ']';
+            '/' + modelName + '[' + idAttr + '=' + paramMap.get('id') + ']';
```
Comment (SeanCondon) [inline src/app/roc-edit-base.ts:163]: how about some fancy template strings?

File: src/app/roc-edit-base.spec.ts [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+});
+const router = jasmine.createSpyObj('router', {
+    initialNavigation: jasmine.createSpy(),
+});
+
+@Component({
+    selector: 'aether-edit-base-spec',
+    template: '<div></div>',
+})
+class RocEditBaseSpecComponent extends RocEditBase<
```
Comment (SeanCondon) [inline src/app/roc-edit-base.spec.ts:63]: nice

File: src/app/aether-slice/slice-edit/slice-edit.component.ts [hunk 5]
```diff
@@ -416,7 +413,7 @@ export class SliceEditComponent
     deleteApplicationFromSelect(app: string): void {
         this.bs.deleteIndexedEntry(
             '/' +
```
Comment (SeanCondon) [inline src/app/aether-slice/slice-edit/slice-edit.component.ts:415]: I think this should have been replaced by full path?

File: src/app/aether-application/application-edit/application-edit.component.ts [hunk 6]
```diff
@@ -139,9 +139,6 @@ export class ApplicationEditComponent
             bs,
             route,
             router,
-            'Enterprises-2.0.0',
```
Comment (SeanCondon) [inline src/app/aether-application/application-edit/application-edit.component.ts:142]: I see you moved these out in to a separate file

File: src/app/roc-data-source.ts [hunk 7]
```diff
@@ -74,9 +74,9 @@ export abstract class RocDataSource<
         protected aetherService: AetherService,
         public bs: BasketService,
         protected target: string,
-        protected pathRoot: string,
-        protected pathListAttr: string[],
-        public indexAttr: string[],
+        protected pathRoot: string, // TODO remove use modelPath
```
Comment (SeanCondon) [inline src/app/roc-data-source.ts:77]: you could use 
```
/** @deprecated */ 
```
before it

**Review by SeanCondon — APPROVED** [review]
LGTM

File: src/app/aether-application/application-edit/application-edit.component.ts (hunk 6 above)
Comment (teone) [reply src/app/aether-application/application-edit/application-edit.component.ts:142]: yes, that way we don't need to duplicate between the `list` and `edit` views. I'm trying to eradicate all the references to the "old" ones.

File: src/app/roc-data-source.ts (hunk 7 above)
Comment (teone) [reply src/app/roc-data-source.ts:77]: as far as I understand you can only deprecate an entire method, not single parameters

**Pull request merged**